### ISSUE TITLE
tx not returned by getAccountTransactions

### ISSUE THREAD

**Issue by friedger** [hirosystems/stacks-blockchain-api#1163]
**Describe the bug**
I query my local node for account transactions for principal `SP000000000000000000002Q6VF78.pox`, however tx `0x39ee4a4423d65e67fcda27c7f09343639d9484bc97a5805b3fbf36d7dc148ecd` is not returned on my local node. Querying a different node the tx is returned.

**To Reproduce**
Not sure whether that is valid for all nodes, for me these two db queries returned different values:

```
select canonical, block_height , * from principal_stx_txs pst where 
principal ='SP000000000000000000002Q6VF78.pox'
and tx_id = '\x39ee4a4423d65e67fcda27c7f09343639d9484bc97a5805b3fbf36d7dc148ecd'
order by block_height desc

canonical|block_height|id     |principal                        |tx_id                                                           |block_height|microblock_sequence|tx_index|canonical|microblock_canonical|
---------+------------+-------+---------------------------------+----------------------------------------------------------------+------------+-------------------+--------+---------+--------------------+
false    |       58941|5769464|SP000000000000000000002Q6VF78.pox|39EE4A4423D65E67FCDA27C7F09343639D9484BC97A5805B3FBF36D7DC148ECD|       58941|         2147483647|       5|false    |true                |
```

```
select canonical, block_height , * from txs pst where 
 tx_id = '\x39ee4a4423d65e67fcda27c7f09343639d9484bc97a5805b3fbf36d7dc148ecd'
order by block_height desc

canonical|block_height|id     |tx_id                                                           |tx_index|raw_result|index_block_hash                                                |block_hash                                                      |block_height|parent_block_hash                                               |burn_block_time|parent_burn_block_time|type_id|anchor_mode|status|canonical|post_conditions|nonce|fee_rate|sponsored|sponsor_address|sender_address                           |origin_hash_mode|event_count|execution_cost_read_count|execution_cost_read_length|execution_cost_runtime|execution_cost_write_count|execution_cost_write_length|raw_tx                                                                                                                                                                                                                                                         |microblock_canonical|microblock_sequence|microblock_hash                                                 |parent_index_block_hash                                         |token_transfer_recipient_address|token_transfer_amount|token_transfer_memo|smart_contract_contract_id|smart_contract_source_code|contract_call_contract_id        |contract_call_function_name|contract_call_function_args                                                               |poison_microblock_header_1|poison_microblock_header_2|coinbase_payload|
---------+------------+-------+----------------------------------------------------------------+--------+----------+----------------------------------------------------------------+----------------------------------------------------------------+------------+----------------------------------------------------------------+---------------+----------------------+-------+-----------+------+---------+---------------+-----+--------+---------+---------------+-----------------------------------------+----------------+-----------+-------------------------+--------------------------+----------------------+--------------------------+---------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+-------------------+----------------------------------------------------------------+----------------------------------------------------------------+--------------------------------+---------------------+-------------------+--------------------------+--------------------------+---------------------------------+---------------------------+------------------------------------------------------------------------------------------+--------------------------+--------------------------+----------------+
true     |       58942|1818467|39EE4A4423D65E67FCDA27C7F09343639D9484BC97A5805B3FBF36D7DC148ECD|       4|0703      |6D25AE78F33DE97D9F219D55AF4C08ADA05A647BB6382E6072956FCD755F77EF|7A9A2E897F9F983CF6F290992361C7D25864338479C77EB59BCC00EC528B03A4|       58942|894847478E8099EB46744EFF08C62C1EEFFE6DEA6E3C3F53F4AAD929DF5D92BE|     1651832067|            1651831416|      2|          3|     1|true     |0200000000     |   69|   86400|false    |               |SP3AQDW78BWS2AZMBZWD7XWGH5HQS3HHVS2MC9NBD|               0|          0|                        9|                     31917|                 57221|                         1|                        131|00000000010400D576F0E85F32257E8BFF1A7EF2112C6F91C63BC8000000000000004500000000000151800001873E8D7721593D9F7249025BBF1777B20223675326374904686BF8D7A8E1BBB75CF0374798BE1B47B2EFF81533A2D93170A3FDC4520133353A20EE172A1A8D830302000000000216000000000000000000000|true                |                  0|1CA0267BC27A14D2B6B3EB87638024DF1E249D9C320862A56F9076BAC5200235|3AEAE038EE6D085F578EFC93D7400438BCAA6885B1337E093A7713FE3C485B70|                                |                     |                   |                          |                          |SP000000000000000000002Q6VF78.pox|delegate-stx               |0000000401000000000000000000000002540BE40005160E0030A7F49ED2E21198215218874716AA0A701A0909|                          |                          |                |
false    |       58941|1818485|39EE4A4423D65E67FCDA27C7F09343639D9484BC97A5805B3FBF36D7DC148ECD|       5|0703      |C3407BE42B4880AB679113F9519AE616A601A7B17E172A0828AC4C51D7404322|8AC0CD0E489B2C6804AFB3698ACD73776EB4EFC228F642D381CB7A9E884FB059|       58941|9278A505371645CBFE91B5443E32F98401D4E441FC33A3E0A9E3F8C77FB81125|     1651831691|            1651830791|      2|          3|     1|false    |0200000000     |   69|   86400|false    |               |SP3AQDW78BWS2AZMBZWD7XWGH5HQS3HHVS2MC9NBD|               0|          0|                        9|                     31917|                 57221|                         1|                        131|00000000010400D576F0E85F32257E8BFF1A7EF2112C6F91C63BC8000000000000004500000000000151800001873E8D7721593D9F7249025BBF1777B20223675326374904686BF8D7A8E1BBB75CF0374798BE1B47B2EFF81533A2D93170A3FDC4520133353A20EE172A1A8D830302000000000216000000000000000000000|true                |         2147483647|                                                                |0CFB16A9CA8B8F7C0007B9D6CF338BFB850C4C86063FE41F4BDF65E3242124E2|                                |                     |                   |                          |                          |SP000000000000000000002Q6VF78.pox|delegate-stx               |0000000401000000000000000000000002540BE40005160E0030A7F49ED2E21198215218874716AA0A701A0909|                          |                          |                |
```

**Expected behavior**
All nodes should return the same list of transactions

**Version**
```json 
{
  "server_version": "stacks-blockchain-api v3.0.1 (master:583c7e4e)",
  "status": "ready",
  "chain_tip": {
    "block_height": 58953,
    "block_hash": "0xbf1a04ee6ae3050a3935f4d6efd8e0a3076106a1ae71c03db1a6464f2d081593",
    "index_block_hash": "0x4e4e0022e7a46be438c4458629b265f66a660149ef34bb179fe67c50bcf59ed1",
    "microblock_hash": "0x5bfdc8e2f11411c71e73c4322b9f57e1df53eb3dfcab4704415a336f94042daf",
    "microblock_sequence": 2
  }
}
```

**Comment by friedger** [comment]
Maybe this PR is related? @rafaelcr  https://github.com/hirosystems/stacks-blockchain-api/pull/1074/files

**Comment by rafaelcr** [comment]
@friedger that's right... this used to be a bug reported in https://github.com/hirosystems/stacks-blockchain-api/issues/1098 and https://github.com/hirosystems/stacks-blockchain-api/issues/1119, and then later on patched by https://github.com/hirosystems/stacks-blockchain-api/pull/1137 for 3.0.3

That PR didn't actually fix the `principal_stx_txs` table, though, that change came later here: https://github.com/hirosystems/stacks-blockchain-api/pull/1135 (to be released in 4.0.0)
